sendermode.go

```diff
@@ -128,38 +132,37 @@ func (fm *Frontman) sendResultsChanToHub(resultsChan *chan Result) error {
 		return fmt.Errorf("postResultsToHub: %s", err.Error())
 	}
 
-	fm.Stats.CheckResultsSentToHub += uint64(len(results))
+	fm.statsLock.Lock()
+	fm.stats.CheckResultsSentToHub += uint64(len(results))
+	fm.statsLock.Unlock()
 
+	fm.offlineResultsLock.Lock()
+	defer fm.offlineResultsLock.Unlock()
 	fm.offlineResultsBuffer = []Result{}
+
 	return nil
 }
 
 // sends results to hub continuously
-func (fm *Frontman) sendResultsChanToHubQueue(resultsChan *chan Result) error {
+func (fm *Frontman) sendResultsChanToHubQueue(interrupt chan struct{}, resultsChan *chan Result) {
```

Comment:
you could use go context instead of interrupt channel

Reply:
Yes, maybe in a future rework. But it is built and tested around using an interrupt channel.